packages/eslint-plugin/src/rules/method-signature-style.ts

```diff
@@ -97,6 +97,17 @@ export default util.createRule<Options, MessageIds>({
       return '';
     }
 
+    function isNodeParentModuleDeclaration(node: any): boolean {
+      if (node.parent?.type === AST_NODE_TYPES.TSModuleDeclaration) {
+        return true;
+      }
+
+      if (node.parent?.type === AST_NODE_TYPES.Program) {
+        return false;
+      }
+      return isNodeParentModuleDeclaration(node.parent);
+    }
```

Comment:
We can remove the `any` here with the catch-all `Node` type

```suggestion
    function isNodeParentModuleDeclaration(node: TSESTree.Node): boolean {
      if (!node.parent) {
        return false;
      }
    
      if (node.parent.type === AST_NODE_TYPES.TSModuleDeclaration) {
        return true;
      }

      if (node.parent.type === AST_NODE_TYPES.Program) {
        return false;
      }
      
      return isNodeParentModuleDeclaration(node.parent);
    }
```